src/TLuaInterpreter.cpp

```diff
@@ -1962,9 +1918,8 @@ int TLuaInterpreter::getLineCount(lua_State* L)
 {
     Host& host = getHostFromLua(L);
     if (lua_isstring(L, 1)) {
-        std::string window = lua_tostring(L, 1);
-        QString _window = window.c_str();
-        lua_pushnumber(L, mudlet::self()->getLastLineNumber(&host, _window) + 1);
+        QString window = lua_tostring(L, 1);
+        lua_pushnumber(L, mudlet::self()->getLastLineNumber(&host, window) + 1);
         return 1;
     } else {
         int lineNumber = host.mpConsole->getLineCount();
```

Comment:
Here's another one.
Also the final `return 0;` seems obsolete

Reply:
I agree about the final "return 0" there; will update the patch.

However I intentionally didn't touch any cases where the "return" does not precede a call to `lua_error`. The reason is that an error call is not symmetric, but some (most?) of the other tests are.

However, there *are* many `return`s which are preceded by "push NIL and an error message" which should be treated like an error; I'll do that also.

Reply:
> However, there are many returns which are preceded by "push NIL and an error message" which should be treated like an error; I'll do that also.

I think this is intentional so the script in lua doesn't break. (I could be wrong)

Reply:
Correct, API design is:

* wrong arguments: error
* correct arguments, but a problem executing it: return nil+msg, do not error

This is standard Lua design

Reply:
Not sure how this error handling influences the if-return-else antipattern.
For me, the else seems superfluous still.

Reply:
I agree, the else is superfluous.

Reply:
Well, `if(foo) return; else bar()` is always equivalent to `if(foo) return; bar()`. That's not the question.

My point is, there's a conceptual difference between returning an error and returning different values depending on the function's arguments. In the error cases (both `lua_error` and `nil-plus-errstr`), the `bar()` should not be indented. However, when we return different values both branches should be recognizeable as equivalent (thus you can swap the branches or add another clause without affecting the indentation), so in that case we should keep the explicit `else`.

Reply:
The `return 0;` is unnecessary - once the `else-after-return` is cleaned up it will be even more clear that it is unreachable code...

On the other hand there is a glaring omission of code (and documentation in the wiki) to handle the case where the `window` argument is supplied but does not exist. Our current paradigm is for the `(int) mudlet::getLastLineNumber(Host*, QString&)` to be revised to return a `std::pair<bool, QString>` where, if there is an error in the deeper code the first argument is a `false` and *in that case* the second argument is a non-translatable string (so will be generated from a `QStringLiteral` which contains a message that may include the `window` value generated in the deeper code) which is then pushed onto the Lua stack preceded with a `nil`. As the `window` name need not contain a Latin1 string we have to use the `QString::toUtf8()` method on the string before we can squirt it back to the Lua subsystem.